programs/uxd/src/utils/math/compute_precision_loss.rs

```diff
@@ -0,0 +1,18 @@
+use anchor_lang::prelude::*;
+
+use crate::error::UxdError;
+
+pub fn compute_precision_loss(total_shares_amount: u64, total_shares_value: u64) -> Result<u64> {
+    // ceil ( total_shares_value / total_shares_amount )
+    // is equivalent to (sum - 1) / divisor + 1
+    if total_shares_value == 0 {
```

Comment:
also should fail if total_shares_supply is null I think:
- we can never compute the price of a LP price if we are missing either the supply or the total value

Reply:
maybe do the same as other compute functions:

```rust
    require!(total_shares_amount > 0, UxdError::MathError);
    require!(total_shares_value > 0, UxdError::MathError);
```

Reply:
renamed it on credix on another PR:
https://github.com/UXDProtocol/uxd-program/pull/223/files#diff-c012d070093436f787024d1947f0a8c45320f25fa52554b7d5aff38112adc76eL14

Reply:
if `total_shares_amount` is zero, the step `total_shares_amount.checked_sub(1)` would abort with MathError already

Reply:
and this zero checking on `total_shares_amount` is sth additional to the original `calculate_possible_lp_token_precision_loss_collateral_value` is shared also to mint/redeem, I doubt should alter the logic here

Reply:
if the `mercurial_vault_lp_mint_supply` is zero, there won't be any effective mint/redeem/profit collection and we do hv a checking on balance change of `collateral_amount` on all three ixs so it's fine anyway?

Reply:
for sure, but this function is supposed to be re-useable in any context, so adding strict require is always a good idea :)

Reply:
ok, added